### ISSUE TITLE
`invalid-attribute-access` - false negative when assigning to non-`ClassVar` on the class

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3721]
### Summary

the example [documented here](https://docs.astral.sh/ty/reference/rules/#invalid-attribute-access) says there should be two errors here:

```py
from typing import ClassVar

class C:
    class_var: ClassVar[int] = 1
    instance_var: int

C.class_var = 3  # okay
C().class_var = 3  # error: Cannot assign to class variable

C().instance_var = 3  # okay
C.instance_var = 3  # error: Cannot assign to instance variable
```

however the second error is not actually reported: https://play.ty.dev/405b2f67-4bd9-4102-9a53-1777c4a45a83

### Version

923717ad3

**Comment by carljm** [comment]
Our behavior here matches mypy/pyright/pyrefly and I'm not convinced we should change it. `instance_var` is not explicitly marked as "should never be set on a class." IMO the documentation should be updated to not assert this error.

**Comment by AlexWaygood** [comment]
Personally I would certainly want an error from my type checker for assigning an instance variable on the class object. I think that's almost always indicative of a mistake.

However, it should probably be a different error code, and possibly not enabled by default. So I agree that to fix this issue, we should just update the docs.

**Comment by DetachHead** [comment]
i agree it should be an error. especially since `invalid-attribute-override` already reports an error when overriding a `ClassVar` with an instance variable -- that's only an issue if treating an instance variable like a `ClassVar` is not allowed

```py
from typing import ClassVar

class Base:
    instance_attr: int
    class_attr: ClassVar[int]

class Sub(Base):
    class_attr: int  # error: [invalid-attribute-override]

Sub.class_attr = 1 # no error
```

[Playground](https://play.ty.dev/2480e980-1584-445e-8249-3c58ad0c33bf)

**Comment by AlexWaygood** [comment]
Apparently we do already emit an error if you assign to an _instance-only_ variable. I forgot that we internally distinguish between _instance_ variables, which can be accessed and set on class objects (for unfortunate backwards-compatibility reasons with the rest of the typing ecosystem), and _instance-only_ variables, which cannot:

```py
class Foo:
    instance_var: int = 42

    def __init__(self):
        self.instance_only_var: int = 42

Foo.instance_var = 56  # fine
print(Foo.instance_var)  # fine

Foo.instance_only_var = 72  # error
print(Foo.instance_only_var)  # error
```

https://github.com/astral-sh/ruff/pull/25863 will fix the examples in the docs here
